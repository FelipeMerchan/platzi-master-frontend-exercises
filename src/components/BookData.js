import React from 'react'

import './styles/BookData.css'

function BookData() {
  return (
    <div className="BookData">
      <h3 className="BookData__title">Datos del libro</h3>
      <ul className="BookData__list">
        <li>
          <span>Título: HTML &amp; CSS:: Curso práctico avanzado.</span>
        </li>
        <li>
          <span>Autor: Sergio Luis Mora</span>
        </li>
        <li>
          <span>Editorial: Publicaciones Altamar</span>
        </li>
        <li>
          <span>Año de publicaci&oacuten: 2015</span>
        </li>
        <li>
          <span>ISBN</span>
        </li>
      </ul>
    </div>
  )
}

export default BookData