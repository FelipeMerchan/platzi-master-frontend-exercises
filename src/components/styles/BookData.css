.BookData__title {
  margin: var(--space) 0;
}

.BookData__list {
  padding-left: 0;
  list-style-type: none;
}

.BookData__list li {
  display: inline;
}

@media screen and (min-width: 767px) {
  .BookData__list {
    padding-left: var(--space-2);
    list-style-type: unset;
  }

  .BookData__list li {
    display: list-item;
  }
}