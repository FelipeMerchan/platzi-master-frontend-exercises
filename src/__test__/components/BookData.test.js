import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import BookData from '../../components/BookData'

describe('Probar el componente <BookData />', () => {
  const bookData = shallow(<BookData />)

  test('Render del componente BookData', () => {
    expect(bookData.length).toEqual(1)
  })
  test('Render del tÃ­tulo', () => {
    expect(bookData.find('.BookData__title').text()).toEqual('Datos del libro')
  })
  test('Render con el className BookData__list', () => {
    expect(bookData.find('ul').hasClass('BookData__list')).toBe(true)
  })
})

describe('BookData Snapshot', () => {
  test('Comprobar la UI del componente BookData', () => {
    const bookData = create(<BookData />)
    expect(bookData.toJSON()).toMatchSnapshot()
  })
})