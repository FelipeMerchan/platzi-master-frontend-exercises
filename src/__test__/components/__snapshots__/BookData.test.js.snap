// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookData Snapshot Comprobar la UI del componente BookData 1`] = `
<div
  className="BookData"
>
  <h3
    className="BookData__title"
  >
    Datos del libro
  </h3>
  <ul
    className="BookData__list"
  >
    <li>
      <span>
        Título: HTML & CSS:: Curso práctico avanzado.
      </span>
    </li>
    <li>
      <span>
        Autor: Sergio Luis Mora
      </span>
    </li>
    <li>
      <span>
        Editorial: Publicaciones Altamar
      </span>
    </li>
    <li>
      <span>
        Año de publicaci&oacuten: 2015
      </span>
    </li>
    <li>
      <span>
        ISBN
      </span>
    </li>
  </ul>
</div>
`;
